File: apps/cargo-tracker/src/main/java/com/example/cargotracker/billing/domain/model/commands/GenerateInvoiceCommand.java
package com.example.cargotracker.billing.domain.model.commands;

public record GenerateInvoiceCommand(String bookingId, String freightChargeId) {
    public GenerateInvoiceCommand {
        if (bookingId == null || bookingId.isBlank()) throw new IllegalArgumentException("予約 ID は必須です");
        if (freightChargeId == null || freightChargeId.isBlank()) throw new IllegalArgumentException("輸送料金 ID は必須です");
    }
}

File: apps/cargo-tracker/src/CargoTracker.Web/Billing/Application/Internal/CommandServices/GenerateInvoiceCommand.cs
namespace CargoTracker.Billing.Application.Internal.CommandServices;

/// <summary>精算書を発行するコマンド（US21/US22）。経理担当者が配送完了予約の料金を算出し割引を適用する。</summary>
public sealed record GenerateInvoiceCommand(
    string BookingId,
    DateTimeOffset IssuedAt,
    int PaymentTermDays = 30);
